nvflare/private/fed/app/simulator/simulator_worker.py

```diff
@@ -0,0 +1,204 @@
+from nvflare.apis.workspace import Workspace
+from nvflare.fuel.common.multi_process_executor_constants import CommunicationMetaData
+from nvflare.fuel.hci.server.authz import AuthorizationService
+from nvflare.fuel.sec.audit import AuditService
+from nvflare.private.defs import EngineConstant
+from nvflare.private.fed.app.client.worker_process import check_parent_alive
+from nvflare.private.fed.client.admin import FedAdminAgent
+from nvflare.private.fed.client.admin_msg_sender import AdminMessageSender
+from nvflare.private.fed.client.client_json_config import ClientJsonConfigurator
+from nvflare.private.fed.client.client_req_processors import ClientRequestProcessors
+from nvflare.private.fed.client.client_run_manager import ClientRunManager
+from nvflare.private.fed.client.client_runner import ClientRunner
+from nvflare.private.fed.client.fed_client import FederatedClient
+from nvflare.private.fed.simulator.simulator_client_engine import SimulatorClientEngine
+from nvflare.security.security import EmptyAuthorizer
+
+
+class ClientTaskWorker(FLComponent):
+    def create_admin_agent(self, server_args, federated_client: FederatedClient, args, rank=0):
+        sender = AdminMessageSender(
+            client_name=federated_client.token,
+            server_args=server_args,
+            secure=False,
+        )
+        client_engine = SimulatorClientEngine(federated_client, federated_client.token, sender, args, rank)
+        admin_agent = FedAdminAgent(
+            client_name="admin_agent",
+            sender=sender,
+            app_ctx=client_engine,
+        )
+        admin_agent.app_ctx.set_agent(admin_agent)
+        federated_client.set_client_engine(client_engine)
+        for processor in ClientRequestProcessors.request_processors:
+            admin_agent.register_processor(processor)
+
+        client_engine.fire_event(EventType.SYSTEM_START, client_engine.new_context())
+
+        return admin_agent
+
+    def create_client_runner(self, client):
+        """Create the ClientRunner for the client to run the ClientApp.
+
+        Args:
+            client: the client to run
+
+        """
+        app_client_root = client.app_client_root
+        args = client.args
+        client_config_file_name = os.path.join(app_client_root, args.client_config)
+        conf = ClientJsonConfigurator(
+            config_file_name=client_config_file_name,
+        )
+        conf.configure()
+        workspace = Workspace(args.workspace, client.client_name, args.config_folder)
+        run_manager = ClientRunManager(
+            client_name=client.client_name,
+            job_id=args.job_id,
+            workspace=workspace,
+            client=client,
+            components=conf.runner_config.components,
+            handlers=conf.runner_config.handlers,
+            conf=conf,
+        )
+        client.run_manager = run_manager
+        with run_manager.new_context() as fl_ctx:
```

Comment:
should you use your class "SimulatorClientAppRunner"?

Reply:
There are some similar codes there, but the logic is not fully the same.

Reply:
refactored